corehq/apps/api/es.py

```diff
@@ -488,19 +494,12 @@ def es_query_from_get_params(search_params, domain, reserved_query_params=None,
     else:
         raise AssertionError("unknown doc type")
 
-    if '_search' in search_params:
+    if '_search' in query_params:
         # This is undocumented usecase by Data export tool and one custom project
         #   Validate that the passed in param is one of these two expected
-        _filter = _validate_and_get_es_filter(json.loads(search_params['_search']))
+        _filter = _validate_and_get_es_filter(json.loads(query_params.pop('_search')))
         query = query.filter(_filter)
 
-    # filters are actually going to be a more common case
-    reserved_query_params = RESERVED_QUERY_PARAMS | set(reserved_query_params or [])
-    query_params = {
-        param: value
-        for param, value in search_params.items()
-        if param not in reserved_query_params and not param.endswith('__full')
```

Comment:
`reserved_query_params` came from three different places - two constants and a param passed in.  Anything remaining in `query_params` by the end of the function is assumed to be a key/value pair stored in ES, and is searched as such.  To function properly, everything else must be removed.  Since `_search` and `include_archived` are specifically handled, I moved the removal of those params next to where they're used, for clarity.